bokeh/models/widgets/widget.py

```diff
@@ -17,3 +17,15 @@ class Widget(LayoutDOM):
     ''' A base class for all interactive widget types.
 
     '''
+
+class NeedsJQuery(object):
+    """ A mixin for widgets requiring JQuery and/or JQuery UI. """
+
+    __javascript__ = [
+        "https://code.jquery.com/jquery-3.2.1.min.js",
+        "https://code.jquery.com/ui/1.12.1/jquery-ui.min.js",
+    ]
+
+    __css__ = [
+        "https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css",
+    ]
```

Comment:
AFAIK date picker is hopelessly broken and has been for a long time. DataTable is very important but probably also the largest widget implementation. How about cleaving it off and bundling JQ-ui with just it? then we can Fix autocomplete and date picker to not use JQ at all.